src/MicrosoftAuthentication.js

```diff
@@ -0,0 +1,36 @@
+const MicrosoftAuthFlow = require('./MicrosoftAuthFlow');
+
+/**
+ * Authenticates with Mincrosoft through user credentials, then
+ * with Xbox Live, Minecraft, checks entitlements and returns profile
+ *
+ * @function
+ * @param {object} client - The client passed to protocol
+ * @param {object} options - Client Options
+ */
+async function authenticatePassword(client, options) {
+    throw Error('Not implemented')
+}
+
+/**
+* Authenticates to Minecraft via device code based Microsoft auth,
+* then connects to the specified server in Client Options
+*
+* @function
+* @param {object} client - The client passed to protocol
+* @param {object} options - Client Options
+*/
+async function authenticateDeviceCode(client, options) {
+    try {
+        const flow = new MsAuthFlow(options.username, options.profilesFolder, options, options.onMsaCode)
+
+        const chain = await flow.getMinecraftToken(client.clientX509)
+        // console.log('Chain', chain)
+        await postAuthenticate(client, options, chain)
```

Comment:
The emissions and postAuth stuff should be separate from the lib, where we're just interested in doing the auth.